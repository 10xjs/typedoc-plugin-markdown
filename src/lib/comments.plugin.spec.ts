import * as Handlebars from 'handlebars';
import * as path from 'path';
import { Application } from 'typedoc';

describe(`CommentsPlugin`, () => {
  const out = path.join(__dirname, '..', 'test', 'out');

  const app = new Application({
    module: 'CommonJS',
    includes: './src/test/fixtures/inc/',
    media: './src/test/fixtures/media/',
    theme: 'markdown',
    plugin: path.join(__dirname, '../../dist/index'),
  });

  const project = app.convert(app.expandInputFiles(['./src/test/examples']));
  app.generateDocs(project, out);

  test(`should define helper'`, () => {
    const helpers = Handlebars.helpers;
    expect(helpers.comment).toBeDefined();
  });

  test(`should convert symbols brackets to symbol links'`, () => {
    expect(go(project.findReflectionByName('commentsWithSymbolLinks').comment.text)).toMatchSnapshot();
  });

  test(`should convert comments with includes'`, () => {
    expect(go(project.findReflectionByName('commentsWithIncludes').comment.text)).toMatchSnapshot();
  });

  function go(text: string) {
    const template = Handlebars.compile('{{{comment}}}');
    return template(text);
  }
});
