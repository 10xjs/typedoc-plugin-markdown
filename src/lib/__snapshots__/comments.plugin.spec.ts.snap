// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsPlugin should convert comments with includes' 1`] = `
"This is a simple example on how to use BaseClass.

![My image alt text](../media/logo.png)

"
`;

exports[`CommentsPlugin should convert symbols brackets to symbol links' 1`] = `
"- Link to a class: [Animal](../classes/_classes_.animal.md)
- Link to a function: [buildName](../modules/_functions_.md#buildname)
"
`;
